### ISSUE TITLE
Partial moving of core objects to interpreter state is incorrect at best, unsafe at worse.

### ISSUE THREAD

**Issue by markshannon** [python/cpython#89854]
BPO | [45691](https://bugs.python.org/issue45691)
--- | :---
Nosy | @vstinner, @markshannon, @ericsnowcurrently, @shihai1991, @erlend-aasland
PRs | <li>python/cpython#29366</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2021-11-02.16:04:49.885>
labels = ['interpreter-core', 'type-bug', '3.11']
title = 'Partial moving of core objects to interpreter state is incorrect at best, unsafe at worse.'
updated_at = <Date 2022-01-18.12:45:01.315>
user = 'https://github.com/markshannon'
```

bugs.python.org fields:
```python
activity = <Date 2022-01-18.12:45:01.315>
actor = 'shihai1991'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Interpreter Core']
creation = <Date 2021-11-02.16:04:49.885>
creator = 'Mark.Shannon'
dependencies = []
files = []
hgrepos = []
issue_num = 45691
keywords = ['patch']
message_count = 5.0
messages = ['405514', '405628', '406483', '410811', '410812']
nosy_count = 5.0
nosy_names = ['vstinner', 'Mark.Shannon', 'eric.snow', 'shihai1991', 'erlendaasland']
pr_nums = ['29366']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue45691'
versions = ['Python 3.11']
```

</p></details>

**Comment by gvanrossum** [comment]
This wasn't closed. Is fixing this still targeted for 3.11? In that case it should probably be made a release blocker. If not, the label should be changed to 3.12. @vstinner @markshannon @ericsnowcurrently

**Comment by vstinner** [comment]
> This wasn't closed.

@markshannon fixed the issue with https://github.com/python/cpython/commit/acc89db9233abf4d903af9a7595a2ed7478fe7d3

This issue can now be closed, no? Or is there a remaining issue?

> I tried once to "free" / reset static types in Py_Finalize(), but it's hard to implement properly :-(

I implemented that in issue #90575.

**Comment by gvanrossum** [comment]
The array of small ints got moved again, it's now in pycore_global_objects.h. I will close this issue.
